File: go-client-sdk/pkg/models/shared/note.go
package shared

type NoteAssociations struct {
	AccountIds  []string `json:"accountIds"`
	ContactIds  []string `json:"contactIds"`
	DealIds     []string `json:"dealIds"`
	LeadIds     []string `json:"leadIds"`
	OwnerUserID string   `json:"ownerUserId"`
}

// Note
// A Note attached to some CRM Object.
type Note struct {
	Additional   map[string]interface{} `json:"additional,omitempty"`
	Associations NoteAssociations       `json:"associations"`
	Content      *string                `json:"content,omitempty"`
	CreatedTime  string                 `json:"createdTime"`
	ID           string                 `json:"id"`
	ModifiedTime string                 `json:"modifiedTime"`
	NativeID     string                 `json:"nativeId"`
}

File: typescript-client-sdk/src/sdk/models/shared/note.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class NoteAssociations extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=accountIds" })
  accountIds: string[];

  @SpeakeasyMetadata({ data: "json, name=contactIds" })
  contactIds: string[];

  @SpeakeasyMetadata({ data: "json, name=dealIds" })
  dealIds: string[];

  @SpeakeasyMetadata({ data: "json, name=leadIds" })
  leadIds: string[];

  @SpeakeasyMetadata({ data: "json, name=ownerUserId" })
  ownerUserId: string;
}


// Note
/** 
 * A Note attached to some CRM Object. 
**/
export class Note extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=additional" })
  additional?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=associations" })
  associations: NoteAssociations;

  @SpeakeasyMetadata({ data: "json, name=content" })
  content?: string;

  @SpeakeasyMetadata({ data: "json, name=createdTime" })
  createdTime: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=modifiedTime" })
  modifiedTime: string;

  @SpeakeasyMetadata({ data: "json, name=nativeId" })
  nativeId: string;
}
